src/app/clusters/service-area-server/service-area-delegate.h

```diff
@@ -0,0 +1,389 @@
+
+constexpr size_t kMaxSizeStatusText = 256;
+
+/**
+ * ServiceArea::Delegate Defines methods for implementing application-specific
+ * logic for the Service Area Cluster.
+ */
+class Delegate
+{
+public:
+    Delegate()          = default;
+    virtual ~Delegate() = default;
+
+    /**
+     * Stop this class objects from being copied.
+     */
+    Delegate(const Delegate &)             = delete;
+    Delegate & operator=(const Delegate &) = delete;
+
+    friend class Instance;
+
+protected:
+    /**
+     * @brief This method will be called during the ServiceArea server initialization after the Instance information has been
+     * validated and the Instance has been registered. This can be used to initialise app logic.
+     */
+    virtual CHIP_ERROR Init() { return CHIP_NO_ERROR; };
+
+    //*************************************************************************
+    // Command handling support
+
+    /**
+     * @brief can the selected locations be set by the client in the current operating mode?
+     * @param[out] statusText text describing why selected locations cannot be set (if return is false).
+     * @return true if the current device state allows selected locations to be set by user.
+     *
+     * @note The statusText field SHOULD indicate why the request is not allowed, given the current mode
+     *       of the device, which may involve other clusters.
+     */
+    virtual bool IsSetSelectedLocationsAllowed(char * statusText) = 0;
+
+    /**
+     * Given a set of locations to be set to the SelectedLocations attribute, this method should check that:
+     *  - There are no duplicates in the list. If there are duplicates, the locationStatus should be set to
+     *     DuplicatedLocations and the statusText should be an empty string.
+     *  - The set of locations as a whole is valid and reachable by the device. If the set of locations is invalid,
+     *     the locationStatus should be set to InvalidSet and the statusText SHALL include a vendor-defined error description.
+     *
+     * The caller of this method will ensure that all the locations in the set are valid supported locations.
+     *
+     * @param[in] req List of new selected locations.
+     * @param[out] locationStatus Success if all checks pass, error code if failure.
+     * @param[out] statusText text describing failure (see description above), size kMaxSizeStatusText + 1 byte for terminating
+     * character.
+     * @param[out] useStatusText if true, the statusText value should be returned in the command response.
+     * @return true if success.
+     *
+     * @note If the SelectLocations command is allowed when the device is operating and the selected locations change to none, the
+     * device must stop.
+     */
+    virtual bool IsValidSelectLocationsSet(const Commands::SelectLocations::DecodableType & req,
+                                           SelectLocationsStatus & locationStatus, char * statusText, bool & useStatusText) = 0;
```

Comment:
statusText should be a MutableCharSpan, no?

Reply:
This has been addressed in [this commit](https://github.com/project-chip/connectedhomeip/pull/33991/commits/38a7e7a9c206391ede4528ffd1bc73c391c400a1).